### DISCUSSION TITLE
How to properly `save` a `.json` file coming from the internet?

### DISCUSSION THREAD

**Discussion by amtoine** [nushell/nushell#6245] [Q&A] [answered]
Hi everyone :wave: :yum: 

## The context
i'm trying to use the [_GitHub_ cli tool](https://github.com/cli/cli) to retrieve information from my _GitHub_ repos :ok_hand: 

for instance, the complete name of my wallpapers repository can be accessed with
```rust
$ gh repo view amtoine/wallpapers --json nameWithOwner
{"nameWithOwner":"amtoine/wallpapers"}
```

## The problem
now, i would like to save this return value in a `.json` format and use it later, e.g. with the `open` command.
```rust
gh repo view amtoine/wallpapers --json nameWithOwner | save result.json
```

However, the format does not look right...
```rust
$ gh repo view amtoine/wallpapers --json nameWithOwner | to json
"{\"nameWithOwner\":\"amtoine/wallpapers\"}\n"
```
and thus, after `| save result.json`, `open result.json` does not give me a nice table and the file is not proper `JSON` :cry: 

## Additional info
i've tried to manually write the `JSON` file and it works :+1: 
```rust
$ {"nameWithOwner": "amtoine/wallpapers"}
╭───────────────┬────────────────────╮
│ nameWithOwner │ amtoine/wallpapers │
╰───────────────┴────────────────────╯
$ {"nameWithOwner": "amtoine/wallpapers"} | to json
{
  "nameWithOwner": "amtoine/wallpapers"
}
$ {"nameWithOwner": "amtoine/wallpapers"} | save foo.json
$ open foo.json
╭───────────────┬────────────────────╮
│ nameWithOwner │ amtoine/wallpapers │
╰───────────────┴────────────────────╯
```

**Comment by kubouch** [comment]
This looks like we might have some escaping error. Can you create an issue for it?

Also, does it happen with other tools than the GitHub CLI? I tried `"{\"nameWithOwner\":\"amtoine/wallpapers\"}\n" | from json` manually and it worked fine.

**Reply by amtoine** [reply]
> Also, does it happen with other tools than the GitHub CLI?
>
i'm quite new to `nushell`, so that's the first time i see this :thinking: 

> `"{\"nameWithOwner\":\"amtoine/wallpapers\"}\n" | from json`
>
it works fine for me too :ok_hand: 

> This looks like we might have some escaping error. Can you create an issue for it?
>
yes i can, let me wrap this up :wink:

**Comment by fdncred** [comment, marked as answer]
If you use `save -r foo.json` it should work. `gh repo view amtoine/wallpapers --json nameWithOwner | save -r result.json`

**Reply by amtoine** [reply]
oooooooh :open_mouth: 
that actually is very smart :star_struck:

**Reply by fdncred** [reply]
I tried to explain the "logic" behind this and why it works this way in the issue you created #6248.

**Reply by amtoine** [reply]
coool thanks :yum:
